cranelift/codegen/src/machinst/vcode.rs

```diff
@@ -504,6 +516,21 @@ impl<I: VCodeInst> VCode<I> {
             let (start, end) = self.block_ranges[block as usize];
             buffer.bind_label(MachLabel::from_block(block));
             label_insn_iix[block as usize] = start;
+
+            // Track BB starts. If we have backed up due to MachBuffer
+            // branch opts, note that the removed blocks were removed.
+            let cur_offset = buffer.cur_offset();
+            if last_offset.is_some() && cur_offset <= last_offset.unwrap() {
+                for i in (0..bb_starts.len()).rev() {
+                    if bb_starts[i].is_some() && cur_offset > bb_starts[i].unwrap() {
+                        break;
+                    }
+                    bb_starts[i] = None;
+                }
+            }
+            bb_starts.push(Some(cur_offset));
```

Comment:
This doesn't exactly correspond with cranelift ir level blocks, right? That was my suggestion. And now that I think about it you may need to use `SecondaryMap` to be able to use the right index.

Reply:
Ah, actually, now that I page this back in, I realize that it probably makes more sense not to try to provide a mapping from Cranelift-level blocks to machine code offsets.

The reason is that blocks can be reordered, sometimes prepended or appended with edge blocks, or elided completely. The intent of this change is to instead provide an independent CFG overlay on the final code that can be used for further analysis; the use-case it's built for (VeriWasm, specifically) doesn't need the correspondence with the CLIF-level CFG.

So I've reverted this back to the `Vec<CodeOffset>`. If we later want to connect `Block`s to `CodeOffset`s we could definitely extend this, but I would want to think about the types and how to make it safer to use first.